### PR TITLE
Resolve deprecations in the tests for `Google Analytics` operators & hook

### PR DESCRIPTION
**Pull request by boraberke**
This PR resolves deprecations of the following operators:

- GoogleAnalyticsDataImportUploadOperator
- GoogleAnalyticsDeletePreviousDataUploadsOperator
- GoogleAnalyticsGetAdsLinkOperator
- GoogleAnalyticsListAccountsOperator  
- GoogleAnalyticsRetrieveAdsLinksListOperator  

and the following hook:
- GoogleAnalyticsHook

related: https://github.com/apache/airflow/issues/38642

### CODE REVIEW

**Review by shahar1 — APPROVED** [review]

**Review by MaksYermak — APPROVED** [review]
LGTM

**Review by eladkal — APPROVED** [review]

File: tests/providers/google/marketing_platform/hooks/test_analytics.py [hunk 1]
```diff
@@ -30,24 +34,31 @@
...
-            delegate_to=DELEGATE_TO,
-            impersonation_chain=IMPERSONATION_CHAIN,
-        )
+        with pytest.warns(AirflowProviderDeprecationWarning, match=DEPRECATION_MESSAGE):
+            hook = GoogleAnalyticsHook(
+                API_VERSION,
+                GCP_CONN_ID,
+                delegate_to=DELEGATE_TO,
+                impersonation_chain=IMPERSONATION_CHAIN,
+            )
```
Comment (eladkal) [inline tests/providers/google/marketing_platform/hooks/test_analytics.py:61]: I am not sure if this is right?
The goal of `TestGoogleAnalyticsHook` is to test the hook so I think this should work directly with `GoogleAnalyticsAdminHook` and we should have a test that make sure if we use `GoogleAnalyticsHook` it raises deprecation warning.

WDYT?

**Review by eladkal — CHANGES_REQUESTED** [review]

File: tests/providers/google/marketing_platform/hooks/test_analytics.py (hunk 1 above)
Comment (boraberke) [reply tests/providers/google/marketing_platform/hooks/test_analytics.py:61]: I believe there is already another class `TestGoogleAnalyticsAdminHook` which tests the `GoogleAnalyticsAdminHook`:
https://github.com/apache/airflow/blob/2149b4dbee8fb524bb235280aaef158afaec8d4a/tests/providers/google/marketing_platform/hooks/test_analytics_admin.py#L43-L47

`TestGoogleAnalyticsHook` will not be needed once `GoogleAnalyticsHook` is deprecated, as the new hook already has its own tests under `TestGoogleAnalyticsAdminHook`. 

Just adding `pytest.warns` (keeping the current imp) to `TestGoogleAnalyticsHook` seems appropriate to me.

WDYT?

Comment (potiuk) [reply tests/providers/google/marketing_platform/hooks/test_analytics.py:61]: Makes sense.

**Review by potiuk — APPROVED** [review]

**Review by eladkal — APPROVED** [review]

**Pull request merged**